### PR TITLE
chore: add support for testing on windows

### PR DESCRIPTION
**Pull request by sramam**
windows compatibility changes from [PR #650](https://github.com/vega/ts-json-schema-generator/pull/650) since the parent's status is uncertain.

### CODE REVIEW

File: test/config.test.ts [hunk 1]
```diff
@@ -51,6 +51,7 @@ function assertSchema(
         const validator = new Ajv({
             // skip full check if we are not encoding refs
             validateFormats: config.encodeRefs === false ? undefined : true,
+            allowUnionTypes: true,
```
Comment (domoritz) [inline test/config.test.ts:54]: Why is this change necessary for windows support?

Comment (sramam) [reply test/config.test.ts]: Sorry - I've been flying fast and loose. Will keep it clean going forward

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -22,7 +27,10 @@ jobs:
       - name: Install Node dependencies
         run: yarn --frozen-lockfile
 
-      - run: yarn lint
+      - name: lint (only on ubuntu)
+        if: ${{ matrix.os !== 'windows-latest' }}
```
Comment (domoritz) [inline .github/workflows/test.yml:31]: Why is linting disabled on windows?

Comment (sramam) [reply .github/workflows/test.yml:31]: I couldn't figure how to resolve the CRLF issues. Since it runs on ubuntu, errors won't creep into the code base.

Comment (domoritz) [reply .github/workflows/test.yml:31]: I fixed it correctly in https://github.com/vega/ts-json-schema-generator/pull/663

Comment (sramam) [reply .github/workflows/test.yml:31]: thank you.

Comment (domoritz) [reply .github/workflows/test.yml:31]: Thanks for sending the pull request.
